File: go/src/misc/fibonacci.go
package main

import (
	"fmt"
)

func main() {
	result := 0
	n := 10

	for i := 0; i <= n; i++ {
		result = fibonacci(i)
		fmt.Printf("fibonacci(%d) = %d\n", i, result)
	}
}

func fibonacci(i int) (fib int) {
	if i == 0 {
		fib = 0
	} else if i == 1 {
		fib = 1
	} else {
		fib = fibonacci(i-1) + fibonacci(i-2)
	}
	return
}

File: scala/functional_programming_in_scala/ch02_exercise01/Fibonacci.scala
object Fibonacci {

  def fib(n: Int): Int = {
    if (n == 0) 0
    else if (n ==1) 1
    else fib(n - 1) + fib(n - 2)
  }

  def main(args: Array[String]): Unit = {
    println(fib(10))
  }
}
